tools/bin/mbedtls-release-checker.py

```diff
@@ -0,0 +1,474 @@
+
+                                    self.logger.log(logging.INFO,
+                                                    "Linking {} as backport of {}".format(target_link_pr.pr_number,
+                                                                                          pr.pr_number))
+
+                                    if not target_link_pr.is_backport:
+                                        target_link_pr.is_backport = True
+                                        self.logger.log(logging.INFO,
+                                                        "Forcing {} to be backport on link".format(target_link_pr.pr_number))
+                                break
+
+    def get_cvs_link(self, link_target, link_text = ''):
+
+        if link_text == '':
+            return '\"=HYPERLINK(\"{}\")\"'.format(link_target)
+        else:
+            return '=HYPERLINK(\"{}\",\"{}\")'.format(link_target, link_text)
+
+
+    def write_csv(self, file_name):
+
+        self.logger.log(logging.INFO, 'Writing CSV file to {}'.format(file_name))
+
+        header_fields = ['Hash', 'PR Number', 'Title', 'Is Backport', 'Has Changelog',
+                         'Changelog Check', 'Backport Check', 'Tests Check', 'Backport Link',
+                         'Linked PRs+']
+
+        with open(file_name, 'w') as csv_file:
+
+            csv_writer = csv.writer(csv_file)
+
+            csv_writer.writerow(header_fields)
+
+            for hash, pr in self.github_prs.items():
+
+                backport_num = pr.backport_of if pr.is_backport else pr.backport
+
+                if backport_num == '':
+                    backport = 'None'
+                else:
+                    backport = self.get_cvs_link('https://github.com/{}/pull/{}'.format(self.github_repo,
+                                                                                        backport_num),
+                                                 backport_num)
+
+                csv_row = [pr.hash, self.get_cvs_link(pr.get_github_link(), pr.pr_number),
+                           pr.title, pr.is_backport, pr.has_changelog, pr.changelog_check,
+                           pr.backport_check, pr.tests_check, backport]
+
+                linked_prs = ''
+                first = True
+
+                for link_pr in pr.linked_prs:
+                    csv_row.append(self.get_cvs_link('https://github.com/{}/pull/{}'.format(self.github_repo,
+                                                                                            link_pr),
+                                                     link_pr))
+
+                csv_writer.writerow(csv_row)
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Attempt to automatically detect missing backports/Changelog entries for an MbedTLS release')
+    parser.add_argument('LTS_range_start', help='Git hash/label of LTS branch range start')
+    parser.add_argument('LTS_range_end', help='Git hash/label of LTS branch range start')
```

Comment:
argparser is using ordereddict so the order of inclusion of the arguments here determines how the will be assigned when parsed. 

For example using the syntax described in the docstring `release-checker.py development-local mbedtls.prev.release mbedtls-lts mbedtls.prev.lts.release` and printing the vars(args) I noticed that: 


~~~~~~
/mbedtls-release-checker.py -vv development-local mbedtls-3.4.1 mbedtls-2.28 v2.28.4
{'LTS_output': None,
 'LTS_range_end': 'mbedtls-3.4.1',
 'LTS_range_start': 'development-local',
 'dev_output': None,
 'dev_range_end': 'v2.28.4',
 'dev_range_start': 'mbedtls-2.28',
 'restricted': False,
 'verbose': False,
 'veryverbose': True}
~~~~~~

Ideally I would adjust the order to be like like that `mbedtls.prev.release` - `mbedtls.dev.branch` - `mbedtls.prev.lts.release` - `mbedtls.lts.branch`

Also I would use the standard long and short notation for all arguments.
e.g.

~~~~~~
def get_latest_tag(lts=False):
    """ Return the latest tags for release and lts releases """

    cmd = ["git","describe","--abbrev=0"]
    cmd += ["origin/mbedtls-2.28"] if lts else ["origin/development"]
    out = subprocess.check_output(cmd) # Add try/catch logic
    return out.decode('UTF-8').strip()

 parser.add_argument("-ltr", "--lts-release", default=get_latest_tag(lts=True), help="Tag pointing to the latest LTS release",`
 parser.add_argument("-ltb", "--lts-branch", default="mbedtls-2.28", help="Branch containing all the code to be included in the new release",
~~~~~~